packages/react/src/components/DatePickerInput/next/DatePickerInput.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  Calendar16,
+  WarningFilled16,
+  WarningAltFilled16,
+} from '@carbon/icons-react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { usePrefix } from '../../../internal/usePrefix';
+
+const DatePickerInput = React.forwardRef(function DatePickerInput(props, ref) {
+  const {
+    datePickerType,
+    disabled = false,
+    helperText,
+    hideLabel,
+    iconDescription,
+    id,
+    invalid = false,
+    invalidText,
+    labelText,
+    onClick = () => {},
+    onChange = () => {},
+    openCalendar,
+    pattern = '\\d{1,2}\\/\\d{1,2}\\/\\d{4}',
+    placeholder,
+    size,
+    type = 'text',
+    warn,
+    warnText,
+    ...rest
+  } = props;
+  const prefix = usePrefix();
+  const datePickerInputProps = {
+    id,
+    onChange: (event) => {
+      if (!disabled) {
+        onChange(event);
+      }
+    },
+    onClick: (event) => {
+      if (!disabled) {
+        onClick(event);
+      }
+    },
+    pattern,
+    placeholder,
+    type,
+  };
+  const wrapperClasses = cx(`${prefix}--date-picker-input__wrapper`, {
+    [`${prefix}--date-picker-input__wrapper--invalid`]: invalid,
+    [`${prefix}--date-picker-input__wrapper--warn`]: warn,
+  });
+  const labelClasses = cx(`${prefix}--label`, {
+    [`${prefix}--visually-hidden`]: hideLabel,
+    [`${prefix}--label--disabled`]: disabled,
+  });
+  const helperTextClasses = cx(`${prefix}--form__helper-text`, {
+    [`${prefix}--form__helper-text--disabled`]: disabled,
+  });
+  const inputClasses = cx(`${prefix}--date-picker__input`, {
+    [`${prefix}--date-picker__input--${size}`]: size,
+    [`${prefix}--date-picker__input--invalid`]: invalid,
+  });
+
+  const label = labelText ? (
+    <label htmlFor={id} className={labelClasses}>
+      {labelText}
+    </label>
+  ) : null;
+
+  const helper = helperText ? (
+    <div className={helperTextClasses}>{helperText}</div>
+  ) : null;
+
+  let error = null;
+  if (invalid) {
+    error = <div className={`${prefix}--form-requirement`}>{invalidText}</div>;
+  } else if (warn) {
+    error = <div className={`${prefix}--form-requirement`}>{warnText}</div>;
+  }
```

Comment:
I think it might be easier to read and maintain these ternarys/conditionals if they were down in `render()`. What do you think?